File: siserp-logistica-frontend/src/main/siserp-logistica-frontend/src/app/modules/intranet/dto/request/compra-registrar.request.ts
import { CompraRegistrarDetalleRequest } from "./compra-registrar-detalle.request";

export class CompraRegistrarRequest {
    idProveedor: number;
    idtEstadoCompra: number;
    idtTipoComprobante: number;
    montoTotal: number;
    serieComprobante: string;
    nroComprobante: string;
    nroOrdenCompra: string
    observacion: string;
    fecha: Date;
    flagActivo: number;
    idUsuarioCrea: number;
    fecUsuarioCrea: Date;
    listaDetalleCompra: CompraRegistrarDetalleRequest[];

    nomProveedor: string;
    tipDocProveedor: number;
    nroDocProveedor: string;
}

File: siserp-logistica-backend/src/main/java/com/besoft/siserp/lgtc/dto/request/CompraRegistrarRequest.java
package com.besoft.siserp.lgtc.dto.request;

import java.io.Serializable;
import java.util.Date;
import java.util.List;

import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

@Getter
@Setter
@ToString
public class CompraRegistrarRequest implements Serializable {
	private static final long serialVersionUID = 1L;

	private Long idProveedor;
	private Long idtEstadoCompra;
	private Long idtTipoComprobante;
	private Double montoTotal;
	private String serieComprobante;
	private String nroComprobante;
	private String nroOrdenCompra;
	private String observacion;
	private Date fecha;
	private Integer flagActivo;
	private Long idUsuarioCrea;
	private Date fecUsuarioCrea;
	private List<CompraRegistrarDetalleRequest> listaDetalleCompra;

	private String nomProveedor;
	private Long tipDocProveedor;
	private String nroDocProveedor;
}
